### PR TITLE
Fix stats for podAndContainerStatsFromCRI

### PR DESCRIPTION
**Pull request by mengjiao-liu**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
According to the feature gate `PodAndContainerStatsFromCRI` design, we should only obtain stats data from CRI when the feature gate `PodAndContainerStatsFromCRI` is enabled. When getting data from CRI fails, we will use cAsvisor stats to populate summary stats information. However, when the current code can obtain stats from CRI, it does not return in time. stats mixed with cAdvisor is still returned.

The commit that introduced this issue:  https://github.com/kubernetes/kubernetes/pull/103095/commits/6b3f8e56622c4d6b1c2ad028b7f8fad5ef2a2e80
in PR: https://github.com/kubernetes/kubernetes/pull/103095

The current unit in the file `pkg/kubelet/stats/cri_stats_provider_test.go` is for the case where there is no test `PodAndContainerStatsFromCRI` for true.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Fix the PodAndContainerStatsFromCRI feature, instead of supplementing with stats from cAdvisor.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2371-cri-pod-container-stats#current-fulfiller-of-metrics-endpoints--future-proposal
```

### CODE REVIEW

File: pkg/kubelet/stats/cri_stats_provider.go [hunk 1]
```diff
@@ -147,6 +147,9 @@ func (p *criStatsProvider) listPodStats(updateCPUNanoCoreUsage bool) ([]statsapi
 			klog.V(5).ErrorS(err,
 				"CRI implementation must be updated to support ListPodSandboxStats if PodAndContainerStatsFromCRI feature gate is enabled. Falling back to populating with cAdvisor; this call will fail in the future.",
 			)
+		} else {
```
Comment (haircommander) [inline pkg/kubelet/stats/cri_stats_provider.go:150]: nit: we can invert these conditions:
```
if err == nil {
    return result, nil
}
s, ok ...
```

**Comment by haircommander** [comment]
:facepalm: thanks @mengjiao-liu ! 
/ok-to-test
/triage accepted

File: pkg/kubelet/stats/cri_stats_provider.go (hunk 1 above)
Comment (mengjiao-liu) [reply pkg/kubelet/stats/cri_stats_provider.go:150]: Of course, I originally wanted to write it this way, but I felt that the code changes would be a bit more. I will update it.

**Comment by mengjiao-liu** [comment]
Since PR #110897 is pending by https://github.com/kubernetes/kubernetes/pull/110897#issuecomment-1301548904, a test for this PR is added first. No more waiting for PR #110897 to be merged.

**Comment by dashpole** [comment]
/approve
cc @bobbypage

**Pull request merged**